docs/DeveloperGuide.adoc

```diff
@@ -176,94 +176,214 @@ through all the stored data and generate the statistics data whenever it is quer
 
-The `redo` command does the opposite -- it calls `Model#redoAddressBook()`, which shifts the `currentStatePointer` once to the right, pointing to the previously undone state, and restores the address book to that state.
+Given below is an example usage scenario and how the display schedule mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time. The `distinctDatesList` will be initialized based on the initial event book state.
+
+image::DisplayScheduleState0.png[]
+
+Step 2. The user executes `add_ev n/Free Cofee ...` to add a new event into the list. The distinctDatesList will be flushed.
+
+image::DisplayScheduleState1.png[]
 
 [NOTE]
-If the `currentStatePointer` is at index `addressBookStateList.size() - 1`, pointing to the latest address book state, then there are no undone address book states to restore. The `redo` command uses `Model#canRedoAddressBook()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the redo.
+If a command that alters the eventBook will cause the distinctDatesList to be flushed. User will be required to input "display_schedule" command again to generate the new list of schedule.
 
-Step 5. The user then decides to execute the command `list`. Commands that do not modify the address book, such as `list`, will usually not call `Model#commitAddressBook()`, `Model#undoAddressBook()` or `Model#redoAddressBook()`. Thus, the `addressBookStateList` remains unchanged.
+Step 3. The user executes `display_schedule` to see all dates and the respective events on those dates. The distinctDateList will be generated again based on the current list of events in the EventList.
 
-image::UndoRedoState4.png[]
+image::DisplayScheduleState2.png[]
 
-Step 6. The user executes `clear`, which calls `Model#commitAddressBook()`. Since the `currentStatePointer` is not pointing at the end of the `addressBookStateList`, all address book states after the `currentStatePointer` will be purged. We designed it this way because it no longer makes sense to redo the `add n/David ...` command. This is the behavior that most modern desktop applications follow.
 
-image::UndoRedoState5.png[]
+Step 4. The user now decides to close the app, the current state of the EventBook and EmployeeBook will be stored, however the DistinctDateList would not.
 
-The following activity diagram summarizes what happens when a user executes a new command:
+image::DisplayScheduleState3.png[]
 
-image::CommitActivityDiagram.png[]
+[NOTE]
+Note that the Display Schedule Feature does not load and store the DistinctDate Objects. It processes and generates the list when it is called upon or when the application starts.
 
-==== Design Considerations
+The following sequence diagram shows how the undo operation works:
 
-===== Aspect: How undo & redo executes
+._Sequence Diagram for DisplaySchedule Command_
```

Comment:
When constructing DisplayScheduleCommand, it's better if you write the constructor call near the arrow as `DisplayScheduleCommand()`. Also, since the instance is named, you should write its named on the return arrow.
<img width="404" alt="Screen Shot 2019-10-24 at 1 47 51 PM" src="https://user-images.githubusercontent.com/44959941/67456956-eaebdf80-f664-11e9-85ff-97daab7cf93b.png">

<img width="274" alt="Screen Shot 2019-10-24 at 1 52 58 PM" src="https://user-images.githubusercontent.com/44959941/67457228-a90f6900-f665-11e9-8a90-dc667ecb9ecd.png">

Should the lifeline of *v* terminate here or did you want to continue the lifeline?